### PR TITLE
Standardize documentation and add Conda configuration for ReadtheDocs 

### PR DESCRIPTION
**Pull request by samjwu**
# Changes

Apply the following changes to documentation on ReadtheDocs:

add version number to documentation left navigation bar and page title
add an "About" section with a license page
enable htmlzip, pdf, epub formats when publishing on Read the Docs
set pdf title, author, copyright, and version
rename .sphinx/.doxygen to sphinx/doxygen
update rocm-docs-core dependency
update dependabot config to include documentation label on PRs

Relates to https://github.com/RadeonOpenCompute/rocm-docs-core/issues/330

# Notes

Runs a custom virtual environment via conda (mambaforge) for ReadtheDocs for doxygen 1.9.8 due to doxygen 1.9.1 (default for Ubuntu 22.04) segfaulting on AMDMIGraphX documentation

### CODE REVIEW

**Comment by samjwu** [comment]
Preview: https://rocm.docs.amd.com/projects/AMDMIGraphX/en/doc-standard/

File: docs/doxygen/Doxyfile [hunk 1]
```diff
@@ -26,7 +26,7 @@ INPUT = ../../src
 
 MACRO_EXPANSION = YES
 
-OUTPUT_DIRECTORY = docBin
+OUTPUT_DIRECTORY = .
```
Comment (pfultz2) [inline docs/doxygen/Doxyfile:29]: The `OUTPUT_DIRECTORY` should be set to a directory in the build directory. Only the final documentation should be in the source directory.

**Review by causten — APPROVED** [review]

**Comment by causten** [comment]
@samjwu can you review Paul's comment and adjust

File: docs/doxygen/Doxyfile (hunk 1 above)
Comment (samjwu) [reply docs/doxygen/Doxyfile]: Addressed in [Place doxygen output in docs build directory](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2472/commits/cb6563d0a45e194f1df1e8f0e2b10ead921963af)

Moves the doxygen output to the documentation build directory

**Comment by samjwu** [comment]
I think the format check is failing because it's trying to apply clang formatting to the Sphinx configuration files (table of contents and requirements.in)

**Pull request merged**